### ISSUE TITLE
Incorrect increment when merging feature branch into master.

### ISSUE THREAD

**Issue by mchippingtonderrick-afl** [GitTools/GitVersion#1689]
When merging a feature branch into Master, GitVersion is incrementing the patch number, rather than the minor version number. The minor version number is correctly incremented when evaluated in-branch, but when merged into Master, it only increments the patch version, as if it was just a commit to master, rather than a merge.

**Customised GitVersion.yml for reference:**
```
mode: ContinuousDelivery
branches:
  master:
    regex: master
    tag: ''
    increment: Patch
  patch:
    regex: '(patch|bugfix|fix)[/-]'
    tag: useBranchName
    increment: Patch
    source-branches: ['master', 'major', 'minor']
  minor:
    regex: '(minor|feature)[/-]'
    tag: useBranchName
    increment: Minor
    source-branches: ['master', 'major']
  major:
    regex: '(major|breaking)[/-]'
    tag: useBranchName
    increment: Major
    source-branches: ['master']
```

**Other potentially relevant information:**
Merge is via merging a pull-request on GitHub.
GitVersion is being used in AzurePipelines triggered on commits, using the GitVersion add-in for it.
Example git history graph for clarity:
```
* - Master - v1.4.6 (calculated, should be v1.5.0)
|\
| * - Feature - v1.5.0-prerelease+2 (calculated)
| * - Feature - v1.5.0-prerelease+1 (calculated)
|/
* - Master - v1.4.5 - (tagged)
```

Any suggestions for what I must be doing wrong here, and how I can make it behave the way I would like it to?

**Comment by mchippingtonderrick-afl** [comment]
Setting the increment for Master to "Inherit" makes it behave correctly in the above situation, but it has since behaved incorrectly in another scenario, using the version increment from an unrelated, and unmerged branch:

```
* - Master - v2.0.0 (calculated, should be v1.5.0)
|\
| * - Feature - v1.5.0-prerelease+2 (calculated)
| * - Feature - v1.5.0-prerelease+1 (calculated)
| | * - Major - v2.0.0-prerelease+1 (calculated)
| |/
|/
* - Master - v1.4.5 - (tagged)
```

These branches both share the same base commit, but I don't see how or why that should cause this.

**Comment by mchippingtonderrick-afl** [comment]
I'd forgotten to post it here, but I have had some further issues with this configuration.

I've tried trimming the set of source-branches to just 'master' on the other branches, so that it wouldn't try to use the wrong branch in the above scenario, however, it gets stuck in an infinite loop of trying to find the increment to inherit, failing to do so, defaulting to using master's increment, and needing to find the increment to inherit again.

I have also tried using Mainline which from what I understand is supposed to be the way to achieve what I'm trying to do here, but I have so far not been able to get it to work correctly.
